src/DIRAC/TransformationSystem/Utilities/ScriptUtilities.py

```diff
@@ -0,0 +1,59 @@
+"""Utilities used by TS scripts."""
+
+from DIRAC import gLogger
+from DIRAC.TransformationSystem.Client.TransformationClient import TransformationClient
+
+
+def _getTransformationID(transName):
+    """Check that a transformation exists and return its ID or None if it doesn't
+    exist.
+
+    :param transName: name or ID of a transformation
+    :type transName: int or string
+
+    :return : transformation ID or None if it doesn't exist
+    """
+    testName = transName
+    trClient = TransformationClient()
+    # We can try out a long range of indices, as when the transformation is not found, it returns None
```

Comment:
Can you elaborate on this comment and the logic with the range loop?

Is this what it does:
You have a transformation named `traf`, you find it, but the status is not in the given list of statuses, so you try `traf-1` and try again? etc. up-to traf-100? This seems at least like a convention not everyone might follow?

Reply:
I am not the original author of this, but IIUC the logic is here because of "derived" transformations (sometimes referred to as "inherited" transformations (see https://dirac.readthedocs.io/en/rel-v8r0/AdministratorGuide/Systems/Transformation/index.html#how-to-derive-a-production). I agree with you that it looks a bit weird...

Reply:
Do the LHCb derived transformations follow this naming convention? I have never used derived transformations so far. At least in the example in the docs this convention is not followed.

Reply:
This `getTransformationName` would be called (e.g.) here
https://github.com/DIRACGrid/DIRAC/blob/07cdfca418ed526a04360e4dcdffa8218d3cd9fc/src/DIRAC/TransformationSystem/Utilities/ReplicationTransformation.py#L65
right?

Reply:
yes

Reply:
Ok, fine by me.
```suggestion
    # Try out a long range of indices, to find any related datamanagement transformation, see ....
```